### DISCUSSION TITLE
How to show SelectOneMenu example?

### DISCUSSION THREAD

**Discussion by ChenYouRen** [primefaces/community#2750] [PrimeFaces] [answered]
Excuse me, my English is not very good.
I see the SelectOneMenu example:
https://www.primefaces.org/showcase/ui/input/oneMenu.xhtml?jfwid=11fab
![螢幕擷取畫面 2024-09-26 100547](https://github.com/user-attachments/assets/ea7e0810-c82f-4c8d-96b4-fba6f7306c4d)
I want to show the text as that "Always Display Label".

But my code show that:
![螢幕擷取畫面 2024-09-26 100810](https://github.com/user-attachments/assets/d0ab7623-d74a-4f8e-ba0e-0c93d2001ac3)
My code:
```
<ui:composition xmlns="http://www.w3.org/1999/xhtml"
                xmlns:ui="http://java.sun.com/jsf/facelets"
                xmlns:h="http://java.sun.com/jsf/html"
                xmlns:p="http://primefaces.org/ui"
                xmlns:f="jakarta.faces.core">
    <f:view locale="#{localeBean.locale}">
        <h:form>
            <p:selectOneMenu id="labeledOption" value="#{localeBean.language}" label="#{text['menu.language']}"
	 	                 alwaysDisplayLabel="true" onchange="submit()">
                <f:selectItem itemValue="en-us" itemLabel="English" />
                <f:selectItem itemValue="zh-tw" itemLabel="繁中" />
            </p:selectOneMenu>
        </h:form>
    </f:view>
</ui:composition>
```
How to fix it?

**Comment by ChenYouRen** [comment]
Wow, when I combine it (`<ui:include>`), it works.
But why?

**Reply by melloware** [reply]
Not sure what you mean combined with ui:include?  Can you post your example?

**Reply by ChenYouRen** [reply]
Yes.
```
<ui:include src="/component/changeLocal.xhtml"/>
```

When I only visit the page: http://localhost:8080/jakartaee-hello-world/component/changeLocal.xhtml
it show: 
![螢幕擷取畫面 2024-09-26 100810](https://github.com/user-attachments/assets/e3163680-b7df-4126-9dde-a7f5a365f82b)

But when I write in and visit the page http://localhost:8080/jakartaee-hello-world/dataTable.xhtml
it show:
![螢幕擷取畫面 2024-09-27 101646](https://github.com/user-attachments/assets/f21c6bb5-941e-4748-90e6-d3a388d8c58c)
```
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml"
      xmlns:p="http://primefaces.org/ui"
      xmlns:h="http://java.sun.com/jsf/html"
      xmlns:ui="http://java.sun.com/jsf/facelets"
      xmlns:f="jakarta.faces.core">
    <h:head>
        <meta charset="UTF-8" />
        <title>Insert title here</title>
    </h:head>
    <h:body>
        <f:view locale="#{localeBean.locale}">
            <div style="text-align: right;">
                <ui:include src="/component/changeLocal.xhtml" />
            </div>
        </f:view>
    </h:body>
</html>
```

I don't why it different.

**Reply by melloware** [reply]
Hmmm yeah that doesn't make sense.

Can you provide an executable example using the [PrimeFaces Test](https://github.com/primefaces/primefaces-test) project.  It is the only way developers can debug your problem to help.

**Reply by ChenYouRen** [reply]
Ok...
Here you are:
[primefaces-test-master.zip](https://github.com/user-attachments/files/17173504/primefaces-test-master.zip)
Use: http://localhost:8080/primefaces-test/test.xhtml
and: http://localhost:8080/primefaces-test/component/changeLocal.xhtml
The SelectOneMenu is different result.

**Reply by melloware** [reply, marked as answer]
I fixed your code:
[discussion-2750.zip](https://github.com/user-attachments/files/17174209/discussion-2750.zip)

![image](https://github.com/user-attachments/assets/fbfabdc5-ae5a-45a2-acd8-1f1af81e36c1)

Your `<f:view>` was in the wrong place. It must wrap the whole page.

**Reply by ChenYouRen** [reply]
Wow, thank you, embarrassed...
It is wrong way which I directly access `/component/changeLocal.xhtml` ?

**Reply by melloware** [reply]
Yes you should never access a ui include directly.
